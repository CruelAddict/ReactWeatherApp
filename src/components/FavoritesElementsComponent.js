import React from 'react'
import {connect} from 'react-redux'
import SideWeatherPanel from '../containers/SideWeatherPanel'

const mapStateToProps = state => ({
    items : state.favorites.items
});

const FavoritesElementComponent = props => {
        return <div className={"favorites-container"}>
            {props.items.map(item => (
                <SideWeatherPanel
                    city={item.name}
                    key={item.id}
                    favoriteId={item.id}
                />
            ))}
            </div>
};

export default connect(mapStateToProps)(FavoritesElementComponent)
