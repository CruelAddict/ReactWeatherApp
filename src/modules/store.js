import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducer from './rootReducer'
import store from "../index";

const savedItems = localStorage.getItem('favorites') ? JSON.parse(localStorage.getItem('favorites')).map(value => {
        return {
            ...value,
            isLoading: true
        };
    }) : [];

const initialState = {
    favorites: {
        items: savedItems
    },
    geo: {
        permissionGranted: null,
        coordinates: null
    }
};

export default () => {
    return createStore(
        reducer,
        initialState,
        applyMiddleware(thunk)
    )
}
