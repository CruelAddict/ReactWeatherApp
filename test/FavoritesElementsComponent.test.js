import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

import FavoritesElementsComponent from '../src/components/FavoritesElementsComponent';
import store from '../src/modules/store'
import Actions from "../src/modules/actions";



describe ('FavoritesElementsComponent', () => {

    it('Renders if store is empty', () => {
        const panel = shallow(
            <FavoritesElementsComponent
                store={store}
            />
        );
        expect(panel).toMatchSnapshot()
    });

    it('Renders if store has items', () => {
        store.dispatch(Actions.createFavorite("London")).then( () => {
            const panel = shallow(
                <FavoritesElementsComponent
                    store={store}
                />
            );
            expect(panel).toMatchSnapshot()
        });
    });

});
