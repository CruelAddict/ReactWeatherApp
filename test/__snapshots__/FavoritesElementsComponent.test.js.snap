// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesElementsComponent Renders if store has items 1`] = `
<ContextProvider
  value={null}
>
  <FavoritesElementComponent
    dispatch={[Function]}
    items={
      Array [
        Object {
          "id": 0,
          "name": "London",
          "pushedToServer": false,
        },
      ]
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`FavoritesElementsComponent Renders if store is empty 1`] = `
<ContextProvider
  value={null}
>
  <FavoritesElementComponent
    dispatch={[Function]}
    items={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
